V1.0.0 update

HusseinElsherbini opened the pull request.

Commit:
added demos for boards using the Wi-Fi  evaluation kit rs9116 and Wi-Fi expansion Boards WF200 and SiWx917. modified generate_metadata.py script to clean folder structure, modified matter.slsdk file to reflect v1.0.0 as new version

Commit:
added missing binaries, regenerated metadata

rerasool reviewed (changes requested):
I found the following discrepancies, not sure if they are intentional.

The following files did not change (I expect them to as it is a newer build):
* Examples/lighting-app-wifi/rs9116/BRD4186C
* Examples/lighting-app-wifi/rs9116/BRD4187C
* Examples/lock-app-wifi/rs9116/BRD4186C
* Examples/lock-app-wifi/rs9116/BRD4187C
* Examples/thermostat-app-wifi/rs9116/BRD4186C
* Examples/thermostat-app-wifi/rs9116/BRD4187C

In fact, looks like we are missing a bunch of boards in our updates for `RS9116` and `WF200`, in this PR's diff, but also in previous update: https://github.com/SiliconLabs/matter_demo/tree/main/Examples/lock-app-wifi/wf200

Commit:
added updated binaries for Lighting-app-wifi,lock-app and thermostat for rs9116 boards BRD4186C and BRD4187C

Commit:
Removed older artifacts for:
• Thread lighting app [BRD4187A]
• Wi-Fi lighting/lock/thermostat app for rs9116/wf200 [BRD4162A-4164A]

Commit:
Removed older artifacts for:
• Thread lighting app [BRD4187A]
• Wi-Fi lighting/lock/thermostat app for rs9116/wf200 [BRD4162A-4164A]
